flink-filesystems/flink-s3-fs-base/pom.xml

```diff
@@ -161,12 +161,17 @@ under the License.
 									<pattern>org.apache.flink.runtime.fs.hdfs</pattern>
 									<shadedPattern>org.apache.flink.fs.s3.common.hadoop</shadedPattern>
 								</relocation>
+								<!-- shade Flink's Hadoop FS utility classes -->
+								<relocation>
+									<pattern>org.apache.flink.runtime.util</pattern>
+									<shadedPattern>org.apache.flink.fs.s3.common</shadedPattern>
+								</relocation>
```

Comment:
I think we don't need to do these changes if we put `HadoopConfigLoader` into the `flink-fs-hadood-shaded` module.